### PR TITLE
feat: add SQL Server JDBC support

### LINKED ISSUE #247: Add JDBC connector support for SQL Server
## Feature Description
Add support for connecting to MS SQL Server instances using the JDBC Socket Factory

## Alternatives Considered
Other ways to connect to a SQL Server instance in Java are to use private IP (which may involve setting up serverless VPC access), manually allowlisting IPs and using public IP, or running the Cloud SQL proxy and connecting to it via TCP. Adding SQL server support to the JDBC socket factory would provide authenticated and encrypted connections natively in Java with minimal developer friction. 

## Additional Context
Waiting for mssql-jdbc maintainers to make a 1 line fix to their driver here (github issue: Microsoft/mssql-jdbc#1382) to implement this. The change was already tested locally and works.

### PR DESCRIPTION
**Pull request by shubha-rajan**
## Change Description

Adds support for using the JDBC Socket Factory with SQL Server instances. Will be ready once microsoft/mssql-jdbc#1420 is merged.

## Checklist

- [x] Make sure to open an issue as a 
  [bug/issue](https://github.com/GoogleCloudPlatform//issues/new/choose) 
  before writing your code!  That way we can discuss the change, evaluate 
  designs, and agree on the general idea.
- [x] Ensure the tests and linter pass
- [ ] Appropriate documentation is updated (if necessary)

## Relevant issues:

- Fixes #247

### CODE REVIEW

File: pom.xml
Comment (kurtisvg) [inline pom.xml:72]: Should we consider stacking these into JDBC and r2dbc folders?

**Review by kurtisvg — APPROVED** [review]

File: pom.xml
Comment (shubha-rajan) [reply pom.xml:72]: That would be a good idea, especially if we plan to add SQL Server r2dbc support soon.

File: jdbc/sqlserver/pom.xml
Comment (kurtisvg) [inline jdbc/sqlserver/pom.xml:13]: We'll have to block on this until we finalize the connector names for now :(

File: jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerIntegrationTests.java
Comment (kurtisvg) [inline jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerIntegrationTests.java:77]: `String.format`?

File: pom.xml
Comment (kurtisvg) [inline pom.xml:64]: This is technically the core for both right? 

That said I think these changes should be in two separate PRs: one to reorganize/rename, and one for jdbc-sqlserver

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerIntegrationTests.java
Comment (shubha-rajan) [reply jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerIntegrationTests.java:77]: fixed. No idea why I did that.

File: pom.xml
Comment (shubha-rajan) [reply pom.xml:64]: I opened #322 to reorganize the folders. Once that's merged, I can rebase this PR.

File: sqlserver/pom.xml
Comment (kurtisvg) [inline sqlserver/pom.xml:16]:
```suggestion
  <name>Cloud SQL JDBC connector for SQL Server</name>
```

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**
